Gradle plugin does not work with plugins DSL

**Describe the bug**
As described in https://plugins.gradle.org/plugin/io.quarkus the preferred way to include the Quarkus Gradle plugin should be
```
plugins {
    id "io.quarkus" version "0.28.0"
}
```

**Expected behavior**
The Gradle plugins API should work for Quarkus.

**Actual behavior**
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'extension-registry'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find org.gradle:gradle-tooling-api:5.5.1.
     Searched in the following locations:
       - https://plugins.gradle.org/m2/org/gradle/gradle-tooling-api/5.5.1/gradle-tooling-api-5.5.1.pom
     Required by:
         project : > io.quarkus:io.quarkus.gradle.plugin:0.28.0 > gradle.plugin.io.quarkus:quarkus-gradle-plugin:0.28.0
```

It works only when using the [legacy plugin syntax](https://plugins.gradle.org/plugin/io.quarkus).

**To Reproduce**
Steps to reproduce the behavior:
1. Create a Gradle project.
2. Add the `"io.quarkus"` plugin as described above.
3. `./gradlew build`

**Environment**
 - Output of `uname -a`: 
```
Linux ws-b83fff23-645e-4022-baec-7cf6c437f8cc 4.14.145+ #1 SMP Tue Oct 8 03:03:11 PDT 2019 x86_64 x86_64 x86_64 GNU/Linux
```
 - Output of `java -version`: 
```
openjdk version "11.0.2" 2019-01-15 LTS
OpenJDK Runtime Environment Zulu11.29+3-CA (build 11.0.2+7-LTS)
OpenJDK 64-Bit Server VM Zulu11.29+3-CA (build 11.0.2+7-LTS, mixed mode)
```
 - Quarkus version: `0.28.0`
 - Gradle version: `5.6.3`

--- a/devtools/gradle/build.gradle
+++ b/devtools/gradle/build.gradle
@@ -27,7 +27,7 @@ dependencies {
     compile "io.quarkus:quarkus-platform-descriptor-resolver-json:${version}"
     compile "io.quarkus:quarkus-development-mode:${version}"
     compile "io.quarkus:quarkus-creator:${version}"
-    compile "org.gradle:gradle-tooling-api:5.5.1"
+    compile gradleApi()
 
     testImplementation 'org.assertj:assertj-core:3.13.2'
     testImplementation 'org.junit.jupiter:junit-jupiter-api:5.5.2'

--- a/devtools/gradle/gradle/wrapper/gradle-wrapper.properties
+++ b/devtools/gradle/gradle/wrapper/gradle-wrapper.properties
@@ -1,5 +1,6 @@
+#Tue Nov 05 11:34:08 BRST 2019
+distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-all.zip
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.6.4-bin.zip
-zipStoreBase=GRADLE_USER_HOME
 zipStorePath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME

